src/main/java/io/fabric8/jenkins/openshiftsync/ImageStreamWatcher.java

```diff
@@ -82,47 +83,59 @@ public synchronized void start() {
-                        }
-                        JenkinsUtils.addPodTemplate(entry.getValue());
-                        trackedImageStreams.put(entry.getKey(), entry.getValue());
+                    for (PodTemplate entry : slavesFromIS) {
+                        if (this.predefinedOpenShiftSlaves.contains(entry.getName()))
+                            continue;
+                        JenkinsUtils.removePodTemplate(entry);
+                        JenkinsUtils.addPodTemplate(entry);
                     }
                     // go back and remove tracked items that no longer are marked slaves
-                    Set<Entry<String, PodTemplate>> entryset = trackedImageStreams.entrySet();
-                    for (Entry<String, PodTemplate> entry : entryset) {
-                        // if the entry is not for the IS from the watch event, skip
-                        if (!entry.getKey().startsWith(isuuid))
+                    Iterator<PodTemplate> iter = JenkinsUtils.getPodTemplates().iterator();
+                    while (iter.hasNext()) {
+                        PodTemplate podTemplate = iter.next();
+                        if (!podTemplate.getName().startsWith(isname))
+                            continue;
+                        if (this.predefinedOpenShiftSlaves.contains(podTemplate.getName()))
                             continue;
-                        if (!slavesFromIS.containsKey(entry.getKey())) {
-                            JenkinsUtils.removePodTemplate(entry.getValue());
-                            entryset.remove(entry); // removes from trackedImageStreams as well
+                        boolean keep = false;
+                        // if an IST based slave, see that particulat tag is still in list
+                        for (PodTemplate entry : slavesFromIS) {
+                            if (entry.getName().equals(podTemplate.getName())) {
+                                keep = true;
+                                break;
+                            }
                         }
+                        if (!keep)
+                            JenkinsUtils.removePodTemplate(podTemplate);
                     }
                     break;
 
                 case DELETED:
-                    entryset = trackedImageStreams.entrySet();
-                    for (Entry<String, PodTemplate> entry : entryset) {
-                        // if the entry is for the IS from the watch event, delete
-                        if (entry.getKey().startsWith(isuuid)) {
-                            JenkinsUtils.removePodTemplate(entry.getValue());
-                            entryset.remove(entry); // removes from trackedImageStreams as well
-                        }
+                    iter = JenkinsUtils.getPodTemplates().iterator();
+                    while (iter.hasNext()) {
+                        PodTemplate podTemplate = iter.next();
+                        if (!podTemplate.getName().startsWith(isname))
```

Comment:
that sounds like it should work, i don't believe you can have colons in your IST or IS names, so there should only ever be one.